src/update-promoted-values.ts

```diff
@@ -150,3 +249,37 @@ function isCollectionPath(value: unknown): value is CollectionPath {
 function isCollectionIndex(value: unknown): value is CollectionIndex {
   return typeof value === 'string' || typeof value === 'number';
 }
+
+async function getRelevantCommits(
+  prevTag: string,
+  nextTag: string,
+  dockerImageRepository: string,
+  repoURL: string,
+  gitHubClient: GitHubClient,
+  dockerRegistryClient: DockerRegistryClient,
+): Promise<RelevantCommit[]> {
+  const commits = await dockerRegistryClient.getGitCommitsBetweenTags({
+    prevTag,
+    nextTag,
+    dockerImageRepository,
+  });
+
+  if (commits.length <= 0) return [];
+
+  const first = commits[0];
+  const last = commits[commits.length - 1];
+
+  const githubCommits = await gitHubClient.compareCommits({
```

Comment:
I don't really understand the logic here. If this function is processing commits that it got from Docker tags then we should just use every commit we found that changes the built version and we don't need to do any more filtering (but we will need to get stuff like title and author... though for a WIP we could just generate GH links for the commit).  If this function is processing gitConfig.ref changes we don't need to look at Docker at all.